core/schema/service.go

```diff
@@ -105,14 +123,87 @@ func (s *Service) Create(ctx context.Context, nsName string, schemaName string,
 		Compatibility: compatibility,
 	}
 	versionID := getIDforSchema(nsName, schemaName, sf.ID)
+	_, prevSchemaData, _ := s.GetLatest(ctx, nsName, schemaName)
 	version, err := s.repo.Create(ctx, nsName, schemaName, mergedMetadata, versionID, sf)
+	changeRequest := &changedetector.ChangeRequest{
+		NamespaceID: nsName,
+		SchemaName:  schemaName,
+		Version:     version,
+		VersionID:   versionID,
+		OldData:     prevSchemaData,
+		NewData:     data,
+		Depth:       s.config.SchemaChange.Depth,
+	}
+	newCtx := context.Background()
+	go s.identifySchemaChange(newCtx, changeRequest)
 	return SchemaInfo{
 		Version:  version,
 		ID:       versionID,
 		Location: fmt.Sprintf("/v1beta1/namespaces/%s/schemas/%s/versions/%d", nsName, schemaName, version),
 	}, err
 }
 
+func (s *Service) identifySchemaChange(ctx context.Context, request *changedetector.ChangeRequest) error {
+	endFunc := s.newrelic.StartGenericSegment(ctx, "Identify Schema Change")
+	defer endFunc()
+	schemaID, err := s.repo.GetSchemaID(ctx, request.NamespaceID, request.SchemaName)
+	if err != nil {
+		log.Printf("got error while getting schema ID from DB %s", err.Error())
+		return err
```

Comment:
Wherever this combination of log.Printf followed by return err, try simplifying it by
```
   return log.Errorf(...)
```

Anyways in the caller, we will be printing and handling it.

Reply:
can not see any log.Errorf. Used errros.New with sprintf

Reply:
my mistake. Try `fmt.Errorf`

Reply:
done